x-pack/plugins/lens/public/indexpattern_datasource/dimension_panel/field_input.tsx

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiFormRow } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React from 'react';
+import { insertOrReplaceColumn } from '../operations/layer_helpers';
+import { FieldSelect } from './field_select';
+import type {
+  FieldInputProps,
+  OperationType,
+  GenericIndexPatternColumn,
+} from '../operations/definitions';
+import type { FieldBasedIndexPatternColumn } from '../operations/definitions/column_types';
+
+export function FieldInput({
+  layer,
+  selectedColumn,
+  columnId,
+  indexPattern,
+  existingFields,
+  operationSupportMatrix,
+  updateLayer,
+  onDeleteColumn,
+  incompleteField,
+  incompleteOperation,
+  incompleteParams,
+  currentFieldIsInvalid,
+  helpMessage,
+  groupId,
+  dimensionGroups,
+  operationDefinitionMap,
+}: FieldInputProps<FieldBasedIndexPatternColumn>) {
+  const selectedOperationDefinition =
+    selectedColumn && operationDefinitionMap[selectedColumn.operationType];
+  // Need to workout early on the error to decide whether to show this or an help text
+  const fieldErrorMessage =
+    ((selectedOperationDefinition?.input !== 'fullReference' &&
+      selectedOperationDefinition?.input !== 'managedReference') ||
+      (incompleteOperation && operationDefinitionMap[incompleteOperation].input === 'field')) &&
+    getErrorMessage(
+      selectedColumn as GenericIndexPatternColumn,
```

Comment:
this type assertion shouldn't be needed.
```suggestion
      selectedColumn,
```